packages/batch/src/BatchProcessor.ts

```diff
@@ -19,7 +19,13 @@ class BatchProcessor extends BasePartialBatchProcessor {
   ): Promise<SuccessResponse | FailureResponse> {
     try {
       const data = this.toBatchType(record, this.eventType);
-      const result = await this.handler(data);
+
+      let result: unknown;
+      if (this.options) {
+        result = await this.handler(data, this.options);
+      } else {
+        result = await this.handler(data);
+      }
```

Comment:
I am curious, why can't we pass undefined options to the handler? The handler provided by the user have to check the options before using, right? Maybe I am missing something. 

In case there needs to be an if-branch based on the options, I'd suggest to simplify to a one line:

```const result = this.options? this.handler(data, this.options) : this.handler(data);```